indico/queries/workflow.py

```diff
@@ -346,7 +346,7 @@ def __init__(
 
     def requests(self):
         if self.files:
-            yield UploadDocument(files=self.files)
+            yield UploadBatched(files=self.files, batch_size=10)
```

Comment:
should batch_size be passed in as a kwarg, default to 10? in case of really large files, even 10 would be too large

Reply:
Yeah that makes sense I can make that change. To be honest I always thought it was more on the user to control the amount and size of files being submitted, and sdk was supposed to just be sufficient to handle 1 large submission

Reply:
right the server can only handle 500mb of files uploaded in 1 req. if a user has 10 60mb files, even if the server can process that (which it can), this SDK call will fail. if we expose the option then the user could set like batch_size 5 and be on their merry way :)

Reply:
:thinking: makes me think mentioning the 500mb limit might be worth doing in the docstring for the batch_size argument, otherwise it's not obvious when/how to use the batch param. like also more efficient for the user to _increase_ batch size if the have 50 really small files

Reply:
> supposed to just be sufficient to handle 1 large submission

for sure. with bundle=True tho, the "1 large submission" could be... large AF

Reply:
> 🤔 makes me think mentioning the 500mb limit might be worth doing in the docstring for the batch_size argument, otherwise it's not obvious when/how to use the batch param. like also more efficient for the user to _increase_ batch size if the have 50 really small files

gotcha i'll add that in as well

Reply:
Eventually we might want to add a little logic to this feature to optimize the batch for users (try to calculate a batch size based on file sizes/counts, or take an overriden one). overall, Astha's feedback I agree with 100% tho :)

Reply:
added batch_size as kwarg

Reply:
dynamic would be sick in the future :)